src/main/java/book/store/controller/OrderController.java

```diff
@@ -0,0 +1,82 @@
+package book.store.controller;
+
+import book.store.dto.order.OrderDto;
+import book.store.dto.order.PlaceOrderRequestDto;
+import book.store.dto.order.UpdateOrderStatusRequestDto;
+import book.store.dto.order.item.OrderItemDto;
+import book.store.model.User;
+import book.store.service.OrderItemService;
+import book.store.service.OrderService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Orders management", description = "Endpoints for orders management")
+@RestController
+@RequestMapping("/api/orders")
+@RequiredArgsConstructor
+public class OrderController {
+    private final OrderItemService orderItemService;
+    private final OrderService orderService;
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    @Operation(summary = "Retrieve user's order history",
+            description = "Retrieve user's order history")
+    public List<OrderDto> getAllOrders(Pageable pageable) {
+        Long userId = getAuthenticatedUser().getId();
+        return orderService.getAllOrders(pageable, userId);
```

Comment:
refactor other method accordingly
```suggestion
    public List<OrderDto> getAllOrders(Pageable pageable, Authentication authentication) {
        User user = (User) authentication.getPrincipal();
        return orderService.getAllOrders(pageable, user.getId());
```